src/Services/PractitionerRoleService.php

```diff
@@ -93,6 +93,7 @@ public function getAll($search = array(), $isAndCondition = true)
             $sqlCondition = ($isAndCondition == true) ? 'AND' : 'OR';
             $sql .= implode(' ' . $sqlCondition . ' ', $whereClauses);
         }
+        $sql .= "GROUP BY prac_role.id";
```

Comment:
yep, was a bit suspicious of these queries not working on the vertical table. Guessing this GROUP BY will need to be by the prac_role.uuid .

Reply:
Just pushed a patch not so confident, please tell me if it doesn't make sense. Testing well :)

Reply:
hi @yashrajbothra , Testing is showing 32 entries when there should only be 6 (on my test data), and a bunch of inaccurate stuff (I think it may be giving every combination of possible role/specialty/user :) ). Using vertical tables (by pivoting them in the query) is definitely tough to learn. In the past, I had to figure this out on some reporting code: https://github.com/openemr/openemr/blob/master/library/report_database.inc#L52-L83
And it was definitely not easy for me to figure it out at that time. So rec keep working on and testing it.

Reply:
Also, will probably be easier to get the query in getOne() done first. Then can leverage that solution for the GetAll().